### PR TITLE
Remove all images and Makefile calls to Python 3.8

### PR DESCRIPTION
**Pull request by daniellutz**
Related to: https://issues.redhat.com/browse/RHOAIENG-12625

As 2024b branch is coming to development, all images based on Python 3.8 will be dropped, as Python 3.8 will only be supported on 2023b branch and for Habana based images that require support for them.

## Description
All Python 3.8 based images need to be removed, as 2024b branch will not provide support anymore, only on branch 2023b.

This will involve the removal of the following images:
- base/anaconda-python-3.8
- base/ubi8-python-3.8
- cuda/ubi8-python-3.8
- habana/1.10.0/ubi8-python-3.8
- habana/1.13.0/ubi8-python-3.8
- jupyter/datascience/anaconda-python-3.8
- jupyter/datascience/ubi8-python-3.8
- jupyter/minimal/ubi8-python-3.8
- runtimes/datascience/ubi8-python-3.8
- runtimes/minimal/ubi8-python-3.8
- runtimes/pytorch/ubi8-python-3.8
- runtimes/tensorflow/ubi8-python-3.8

Also the image calls need to be removed from the Makefile, the CI matrix and any other places that reference to Python 3.8.

## How Has This Been Tested?
- Removal of all images based on and Makefile calls to Python 3.8

## Merge criteria:
- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] All Python 3.8 references and dependencies have been correctly removed

### CODE REVIEW

**Comment by atheo89** [comment]
Thank you Daniel for this clean up!! :slightly_smiling_face:
Could you remove also from this [workbenches.md](https://github.com/daniellutz/notebooks/blob/main/docs/workbenches.md) file the ubi8 references and the habana ones? 

Otherwise /lgtm

File: Makefile [hunk 1]
```diff
@@ -614,8 +501,6 @@ test-%: bin/kubectl
 		$(call test_with_papermill,intel/ml,ubi9,python-$(PYTHON_VERSION)) \
 	elif echo "$(FULL_NOTEBOOK_NAME)" | grep -q "trustyai-ubi9"; then \
 		$(call test_with_papermill,trustyai,ubi9,python-$(PYTHON_VERSION)) \
-	elif echo "$(FULL_NOTEBOOK_NAME)" | grep -q "minimal-ubi8"; then \
-		$(call test_with_papermill,minimal,ubi8,python-3.8) \
 	elif echo "$(FULL_NOTEBOOK_NAME)" | grep -q "datascience-ubi8"; then \
 		$(MAKE) validate-ubi8-datascience -e FULL_NOTEBOOK_NAME=$(FULL_NOTEBOOK_NAME); \
```
Comment (jstourac) [inline Makefile:505]: Hmm, what about these 2 lines? I think we should remove them too.
```suggestion
```

**Comment by jstourac** [comment]
On top of what Adriana proposed, maybe we could also fix proper version in these:
```
cuda/ubi9-python-3.9/Dockerfile:      description="CUDA Python 3.9 builder image based on UBI8 for ODH notebooks" \
cuda/ubi9-python-3.9/Dockerfile:      io.k8s.description="CUDA Python 3.9 builder image based on UBI8 for ODH notebooks" \
cuda/ubi9-python-3.11/Dockerfile:      description="CUDA Python 3.11 builder image based on UBI8 for ODH notebooks" \
cuda/ubi9-python-3.11/Dockerfile:      io.k8s.description="CUDA Python 3.11 builder image based on UBI8 for ODH notebooks" \
```
But since there is a discussion whether we shouldn't remove the 3.9 from this branch also, so probably not worth of any effort.

Moreover, there are still some other references to UBI8, but it's probably not that important as it's usually used as an example value in our docs.

So, feel free to ignore this comment :grinning:

**Review by jstourac — COMMENTED** [review]
/lgtm

**Comment by atheo89** [comment]
/override ci/prow/runtimes-ubi8-e2e-tests
/override ci/prow/runtime-pytorch-ubi8-python-3-8-pr-image-mirror
/override ci/prow/runtime-minimal-ubi8-python-3-8-pr-image-mirror
/override ci/prow/runtime-datascience-ubi8-python-3-8-pr-image-mirror
/override ci/prow/runtime-cuda-tensorflow-ubi8-python-3-8-pr-image-mirror
/override ci/prow/notebooks-ubi8-e2e-tests
/override ci/prow/notebook-jupyter-minimal-ubi8-python-3-8-pr-image-mirror
/override ci/prow/notebook-jupyter-datascience-ubi8-python-3-8-pr-image-mirror

**Comment by atheo89** [comment]
/override ci/prow/anaconda-ubi8-e2e-tests
/override ci/prow/habana-notebooks-e2e-tests
/override ci/prow/images 
/override ci/prow/jupyter-datascience-anaconda-python-3-8-pr-image-mirror 
/override ci/prow/notebook-base-ubi8-python-3-8-pr-image-mirror
/override ci/prow/notebook-cuda-jupyter-ds-ubi8-python-3-8-pr-image-mirror 
/override ci/prow/notebook-cuda-jupyter-ds-ubi8-python-3-8-pr-image-mirror 
/override ci/prow/notebook-cuda-jupyter-minimal-ubi8-python-3-8-pr-image-mirror 
/override ci/prow/notebook-cuda-jupyter-minimal-ubi8-python-3-8-pr-image-mirror
/override ci/prow/notebook-habana-1-10-0-ubi8-python-3-8-pr-image-mirror 
/override ci/prow/notebook-habana-1-13-0-ubi8-python-3-8-pr-image-mirror

**Comment by atheo89** [comment]
/override ci/prow/notebook-cuda-ubi8-python-3-8-pr-image-mirror

**Comment by jstourac** [comment]
> Could you remove also from this [workbenches.md](https://github.com/daniellutz/notebooks/blob/main/docs/workbenches.md) file the ubi8 references and the habana ones?

JFYI - I think that this one wasn't updated.

**Pull request merged**